src/xtgeo/grid3d/_grid3d_fence.py

```diff
@@ -31,15 +52,119 @@ def get_randomline(
     hincrement: float | int | None = None,
     atleast: int = 5,
     nextend: int = 2,
+) -> tuple[float, float, float, float, np.ndarray]:
+    import os
+
+    # keep the old method as hidden for reference since it proves to be faster in some
+    # real cases. Need to fully understand this to optimise the new, but will postpone
+    # this for now. TODO: remove this after a while when the new version is fully tested
+    if os.environ.get("XTGEO_RANDOMLINE_OLD", None):
```

Comment:
As the comment indicates; keep the "old" method while doing further optimizing. In units tests the new method is faster but on some real fields with many cells, the "old" is ~3 times faster for unknown reasons.